spec/controllers/articles_controller_spec.rb

```diff
@@ -37,76 +30,83 @@
   end
 
   describe 'POST #create' do
-    context 'with correct params' do
+    context 'when user is logged in' do
+      context 'with correct params' do
 
-      let(:article) { create(:article) }
+        let(:article) { create(:article) }
 
-      let(:params) { {article: {title: article.title,
-                                content: article.content}} }
+        let(:params) { {article: {title: article.title,
+                                  content: article.content}} }
 
-      before(:each) do
-        allow(controller).to receive(:current_user).and_return(article.user)
-      end
+        before(:each) do
+          allow(controller).to receive(:current_user).and_return(article.user)
+        end
 
-      it 'saves a new article to the database' do
-        expect {
+        it 'saves a new article to the database' do
+          expect {
+            post :create, params: params
+          }.to change(Article, :count).by(1)
+        end
+
+        it 'displays a success message' do
           post :create, params: params
-        }.to change(Article, :count).by(1)
-      end
+          expect(response.request.flash[:success]).to_not be_nil
+        end
 
-      it 'displays a success message' do
-        post :create, params: params
-        expect( response.request.flash[:success] ).to_not be_nil
+        it 'redirects to homepage' do
+          post :create, params: params
+          expect(response).to redirect_to root_path
+        end
       end
 
-      it 'pending: displays the article post'
+      context 'without a title included in params' do
+        let(:article) { create(:article) }
```

Comment:
You should specify the `title` as `nil` here instead of changing it on lines 69 and 70:
```suggestion
        let(:article) { create(:article, title: nil) }
```

Reply:
This breaks the flow which I outlined in the previous PR #10 [here](https://github.com/meganduffy/onbording-project/pull/10#discussion_r397953989).

Perhaps I can create a separate factory with an incomplete article? The method above works as expected however.

Reply:
In the link you referenced the variable was a hash to be posted to the controller, which was built using information from the article model, and that is indeed correct. But here you are associating a new variable to a model object directly, just to modify it later. My point here is to create it the way you need straight away.

Creating a separate factory for an incomplete article is a good practice, you could do it just as an exercise, for I think the suggestion above doesn't increase complexity or hampers readability.

Reply:
When the set up is changed to create the article with the title set to nil, the tests fail due to the model validation. 

```
Failures:

  1) ArticlesController POST #create when user is logged in without a title included in params does not save a new article to the database
     Failure/Error: let(:article) { create(:article, title: nil) }
     
     ActiveRecord::RecordInvalid:
       Validation failed: Title can't be blank
     # /Users/megan.duffy/.rvm/gems/ruby-2.6.3/gems/factory_bot-5.1.1/lib/factory_bot/evaluation.rb:18:in `create'
     # /Users/megan.duffy/.rvm/gems/ruby-2.6.3/gems/factory_bot-5.1.1/lib/factory_bot/strategy/create.rb:12:in `block in result'
     # /Users/megan.duffy/.rvm/gems/ruby-2.6.3/gems/factory_bot-5.1.1/lib/factory_bot/strategy/create.rb:9:in `tap'
     # /Users/megan.duffy/.rvm/gems/ruby-2.6.3/gems/factory_bot-5.1.1/lib/factory_bot/strategy/create.rb:9:in `result'
     # /Users/megan.duffy/.rvm/gems/ruby-2.6.3/gems/factory_bot-5.1.1/lib/factory_bot/factory.rb:43:in `run'
     # /Users/megan.duffy/.rvm/gems/ruby-2.6.3/gems/factory_bot-5.1.1/lib/factory_bot/factory_runner.rb:29:in `block in run'
     # /Users/megan.duffy/.rvm/gems/ruby-2.6.3/gems/factory_bot-5.1.1/lib/factory_bot/factory_runner.rb:28:in `run'
     # /Users/megan.duffy/.rvm/gems/ruby-2.6.3/gems/factory_bot-5.1.1/lib/factory_bot/strategy_syntax_method_registrar.rb:20:in `block in define_singular_strategy_method'
     # ./spec/controllers/articles_controller_spec.rb:63:in `block (5 levels) in <main>'
     # ./spec/controllers/articles_controller_spec.rb:66:in `block (5 levels) in <main>'

  2) ArticlesController POST #create when user is logged in without a title included in params re-renders #new
     Failure/Error: let(:article) { create(:article, title: nil) }
     
     ActiveRecord::RecordInvalid:
       Validation failed: Title can't be blank
     # /Users/megan.duffy/.rvm/gems/ruby-2.6.3/gems/factory_bot-5.1.1/lib/factory_bot/evaluation.rb:18:in `create'
     # /Users/megan.duffy/.rvm/gems/ruby-2.6.3/gems/factory_bot-5.1.1/lib/factory_bot/strategy/create.rb:12:in `block in result'
     # /Users/megan.duffy/.rvm/gems/ruby-2.6.3/gems/factory_bot-5.1.1/lib/factory_bot/strategy/create.rb:9:in `tap'
     # /Users/megan.duffy/.rvm/gems/ruby-2.6.3/gems/factory_bot-5.1.1/lib/factory_bot/strategy/create.rb:9:in `result'
     # /Users/megan.duffy/.rvm/gems/ruby-2.6.3/gems/factory_bot-5.1.1/lib/factory_bot/factory.rb:43:in `run'
     # /Users/megan.duffy/.rvm/gems/ruby-2.6.3/gems/factory_bot-5.1.1/lib/factory_bot/factory_runner.rb:29:in `block in run'
     # /Users/megan.duffy/.rvm/gems/ruby-2.6.3/gems/factory_bot-5.1.1/lib/factory_bot/factory_runner.rb:28:in `run'
     # /Users/megan.duffy/.rvm/gems/ruby-2.6.3/gems/factory_bot-5.1.1/lib/factory_bot/strategy_syntax_method_registrar.rb:20:in `block in define_singular_strategy_method'
     # ./spec/controllers/articles_controller_spec.rb:63:in `block (5 levels) in <main>'
     # ./spec/controllers/articles_controller_spec.rb:66:in `block (5 levels) in <main>'

Finished in 0.43421 seconds (files took 2.05 seconds to load)
40 examples, 2 failures
```

I'm curious to see if an incomplete article factory would have the same difficulties. 

I'm not too sure I understand what suggestion you're referring to when you say the following:

> I think the suggestion above doesn't increase complexity or hampers readability.

Reply:
Would you be able to clarify?

Reply:
I realised I missed the point here, as you are making the same use as you referenced earlier. I'm sorry for the confusion.

But trying to clarify the creation of a different factory, if you want different setups for factories you could go and change the properties in the test files or create different factories. The former might be more simple but it can end up "polluting" the code if there are too many occurrences, while the later will be cleaner, more optimised. Ideally you should only optimise your code as needed, not just for the sake of optimisation.